lighthouse-core/report/html/renderer/psi.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * @license
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/* globals self DOM PerformanceCategoryRenderer Util DetailsRenderer */
+
+class PSI {
+  /**
+   * Returns all the elements that PSI needs to render the report
+   * We keep expose this helper method to minimize the 'public' API surface of the renderer
+   * and allow us to refactor without two-sided patches.
+   *
+   *   const {scoreGaugeEl, perfCategoryEl, finalScreenshotDataUri} = PSI.prepareLabData(
+   *      LHresponseJsonString,
+   *      document
+   *   );
+   *
+   * @param {string} LHresponseJsonString
+   * @param {Document} document
+   * @return {{scoreGaugeEl: Element, perfCategoryEl: Element, finalScreenshotDataUri: string|null}}
+   */
+  static prepareLabData(LHresponseJsonString, document) {
+    const lhResult = /** @type {LH.Result} */ JSON.parse(LHresponseJsonString);
+    const dom = new DOM(document);
+
+    const reportLHR = Util.prepareReportResult(lhResult);
+    const perfCategory = reportLHR.reportCategories.find(cat => cat.id === 'performance');
+    if (!perfCategory) throw new Error(`No performance category. Can't make lab data section`);
+    if (!reportLHR.categoryGroups) throw new Error(`No category groups found.`);
+
+    const perfRenderer = new PerformanceCategoryRenderer(dom, new DetailsRenderer(dom));
+    const perfCategoryEl = perfRenderer.render(perfCategory, reportLHR.categoryGroups);
+
+    const scoreGaugeEl = dom.find('.lh-score__gauge', perfCategoryEl);
+    const scoreGaugeWrapper = dom.find('.lh-gauge__wrapper', scoreGaugeEl);
+    scoreGaugeWrapper.classList.add('lh-gauge__wrapper--huge');
+    // Remove Performance category title/description
+    dom.find('.lh-category-header', perfCategoryEl).remove();
+    // Remove navigation links
+    scoreGaugeWrapper.removeAttribute('href');
+    dom.find('.lh-permalink', perfCategoryEl).remove();
```

Comment:
these `remove()`s are definitely the hackiest bits. Could we do with CSS or have a method for it on `CategoryRenderer` or something else?

Reply:
impl-wise it really comes down to [avoiding these two lines](https://github.com/GoogleChrome/lighthouse/blob/e9d9fbf8b304473148462e787f77c9118b9acb91/lighthouse-core/report/html/renderer/performance-category-renderer.js#L120-L121), so yah we can easily avoid these .remove's.

1. option: add extra arg to PerfCatRen.render() that avoids making the permamlink & catHeader elements in the first place. 1 more arg to .render() than the base class, but shrug.
1. option: just like `this._dom.isDevTools()` we add a `this._dom.isPSI()` and drop it into PCR.render().
1. the CSS solution (`display:none`) is possible and technically safe for screenreaders, but it feels a tad messy.
1. what kinda method on CatRen are you thinking of?

Reply:
> option: just like `this._dom.isDevTools()` we add a `this._dom.isPSI()` and drop it into `PCR.render()`.

this has its downsides, but since we have precedent with `isDevTools()` it seems (to me) like the best option out of these (besides a `.isPSI` CSS class, but seems like you really don't like that :)

> what kinda method on CatRen are you thinking of?

I was hoping if I suggested it vaguely enough you'd assume I meant something in particular and bring that back as a possible solution :)